hs_core/views/resource_access_api.py

```diff
@@ -0,0 +1,182 @@
+
+from rest_framework.response import Response
+from rest_framework import generics, serializers
+from rest_framework import status
+
+from hs_core import hydroshare
+from hs_core.hydroshare import utils
+from hs_access_control.models import UserResourcePrivilege, GroupResourcePrivilege, \
+    PrivilegeCodes, UserAccess
+from hs_core.views import utils as view_utils
+from hs_core.views.utils import ACTION_TO_AUTHORIZE
+
+
+class PrivilegeField(serializers.Field):
+    def to_representation(self, privilege):
+        return PrivilegeCodes.PRIVILEGE_CHOICES[privilege-1][1]
+
+
+class GroupResourcePrivilegeSerializer(serializers.ModelSerializer):
+    privilege = PrivilegeField()
+
+    class Meta:
+        model = GroupResourcePrivilege
+        fields = ('id', 'privilege', 'group', 'resource', 'grantor')
+
+
+class UserResourcePrivilegeSerializer(serializers.ModelSerializer):
+    privilege = PrivilegeField()
+
+    class Meta:
+        model = UserResourcePrivilege
+        fields = ('id', 'privilege', 'user', 'resource', 'grantor')
+
+
+class ResourceAccessUpdateDelete(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Read, update, or delete a resource
```

Comment:
Probably this comment be "Read, update or delete access permission for a resource"